src/lib/y2storage/boot_requirements_checker.rb

```diff
@@ -148,6 +150,17 @@ def arch_strategy_class
       end
     end
 
+    # Whether BLS boot will be used for booting
+    #
+    # @return [Boolean]
+    def bls?
+      Yast.import "Linuxrc"
```

Comment:
We prefer to use environment variables instead of directly reading Linuxrc arguments. For several reasons that include from easier testing and development to avoiding coupling the library to Linuxrc (for example, in SLFO this library is still used but Linuxrc is not there).

See how some env variables are recognized at https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/storage_env.rb and then used in other parts of the code. There is even a helper that will turn the variable into boolean for you checking for "1", "on" and all the usual Linuxrc ways of handling that.

If I'm not mistaken, if you use a variable name that starts with `YAST_` that will automagically be available to YaST as environment variable and it will not be persisted in the installed system. So no modification to linuxrc would be needed.

Reply:
Good point. Espl. regarding the future of linuxrc. Fixed.